src/nomad_simulations/outputs.py

```diff
@@ -15,38 +15,153 @@
+            - Directly parsed: the output property is directly parsed from the simulation output files.
+            - Derived: the output property is derived from other output properties. No extra numerical settings
+                are required to calculate the output property.
+            - Post-processing: the output property is derived from other output properties. Extra numerical settings
+                are required to calculate the output property.
+        """,
+    )
+
+    outputs_ref = Quantity(
+        type=Reference(SectionProxy('BaseOutputs')),
+        description="""
+        Reference to the `BaseOutputs` section from which the output property was derived. This is only
+        relevant if `is_derived` is set to True.
+        """,
+    )
+
+    def check_is_derived(self, is_derived: bool, outputs_ref) -> Optional[bool]:
+        if not is_derived:
+            if outputs_ref is not None:
+                return True
+            return False
+        elif is_derived and outputs_ref is not None:
+            return True
+        return None
+
     def normalize(self, archive, logger) -> None:
         super().normalize(archive, logger)
-        self.logger = logger
+
+        check_derived = self.check_is_derived(self.is_derived, self.outputs_ref)
+        if check_derived is not None:
+            self.is_derived = check_derived
+        else:
+            logger.error(
+                'A derived output property must have a reference to another `Outputs` section.'
+            )
+            return
+
+
+class Outputs(BaseOutputs):
+    """
+    Output properties of a simulation.
+
+    # ! add more description once we defined the output properties
+    """
+
+    n_scf_steps = Quantity(
+        type=np.int32,
+        description="""
+        Number of self-consistent steps to converge the output property.
+        """,
+    )
+
+    scf_step = SubSection(
+        sub_section=BaseOutputs.m_def,
+        repeats=True,
+        description="""
+        Self-consistent (SCF) steps performed for converging a given output property.
+        """,
+    )
+
+    is_converged = Quantity(
+        type=bool,
+        description="""
+        Flag indicating whether the output property is converged or not. This quantity is connected
+        with `SelfConsistency` defined in the `numerical_settings.py` module.
+        """,
+    )
+
+    self_consistency_ref = Quantity(
+        type=SelfConsistency,
+        description="""
+        Reference to the `SelfConsistency` section that defines the numerical settings to converge the
+        output property.
+        """,
+    )
+
+    # ? Can we add more functionality to automatically check convergence from `self_consistency_ref` and the last `scf_step[-1]`
+    def check_is_converged(self, is_converged: bool, logger: BoundLogger) -> bool:
+        if not is_converged:
+            logger.info('The output property is not converged.')
```

Comment:
Lots of people ask me for these kinds of warnings, so I'm glad you added them.
I just fear that it might end up getting buried in `info`. We should look into a better semantic category.

Reply:
Well, `is_scf_converged` will help on this 😉

Reply:
No, the requests are more generic than just SCF. Kind of, raise a flag when some setting is inappropriate.

Reply:
Yeah, indeed. We should use some other category, or some functionality to extract and use logger messages (like the one Sarthak did for some testing).

Anyways, I leave a question mark here for the time being.